project/tenantconfig.go

```diff
@@ -0,0 +1,182 @@
+package project
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/suborbital/appspec/tenant"
+	"github.com/suborbital/subo/subo/util"
+)
+
+// WriteTenantConfig writes a tenant config to disk.
+func WriteTenantConfig(cwd string, cfg *tenant.Config) error {
+	filePath := filepath.Join(cwd, "tenant.json")
+
+	configBytes, err := cfg.Marshal()
+	if err != nil {
+		return errors.Wrap(err, "failed to Marshal")
+	}
+
+	if err := ioutil.WriteFile(filePath, configBytes, util.PermFilePrivate); err != nil {
+		return errors.Wrap(err, "failed to WriteFile")
+	}
+
+	return nil
+}
+
+// readTenantConfig finds a tenant.json from disk but does not validate it.
+func readTenantConfig(cwd string) (*tenant.Config, error) {
+	filePath := filepath.Join(cwd, "tenant.json")
+
+	tenantBytes, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to ReadFile for Directive")
+	}
+
+	tenant := &tenant.Config{}
+	if err := tenant.Unmarshal(tenantBytes); err != nil {
+		return nil, errors.Wrap(err, "failed to Unmarshal Directive")
+	}
+
+	return tenant, nil
+}
+
+// readQueriesFile finds a queries.yaml from disk.
+func readQueriesFile(cwd string) ([]tenant.DBQuery, error) {
+	filePath := filepath.Join(cwd, "Queries.yaml")
+
+	configBytes, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, nil
+		}
+
+		return nil, errors.Wrap(err, "failed to ReadFile for Queries.yaml")
+	}
+
+	tenant := &tenant.Config{}
+	if err := tenant.Unmarshal(configBytes); err != nil {
+		return nil, errors.Wrap(err, "failed to Unmarshal Directive")
+	}
+
+	return tenant.DefaultNamespace.Queries, nil
+}
+
+// AugmentAndValidateModules ensures that all modules referenced in a workflow exist
+// in the project and then adds the module list to the provided config.
+func AugmentAndValidateModules(cfg *tenant.Config, mods []ModuleDir) error {
```

Comment:
Is there a better name we could come up with for this? Seems like it is mostly adding revisions.

Reply:
I changed it to `CalculateModuleRefs`